packages/babel-cli/src/babel/dir.ts

```diff
@@ -49,16 +49,22 @@ export default async function ({
 
       if (!res) return FILE_TYPE.IGNORED;
 
-      // we've requested explicit sourcemaps to be written to disk
-      if (
-        res.map &&
-        babelOptions.sourceMaps &&
-        babelOptions.sourceMaps !== "inline"
-      ) {
-        const mapLoc = dest + ".map";
-        res.code = util.addSourceMappingUrl(res.code, mapLoc);
-        res.map.file = path.basename(relative);
-        outputFileSync(mapLoc, JSON.stringify(res.map));
+      if (res.map) {
+        let outputMap: true | false | "both" = false;
```

Comment:
Nit: it would be easier to follow the logic if we use `"external"` instead of `true` (we cannot change the public API, but the internal types yes)